python/src/wslink/protocol.py

```diff
@@ -262,7 +266,10 @@ async def onMessage(self, is_binary, msg, client_id):
                 await self.onCompleteMessage(full_message, client_id)
 
     async def onCompleteMessage(self, rpc, client_id):
-        logger.debug("wslink incoming msg %s", self.payloadWithSecretStripped(rpc))
+        debug_message = "wslink incoming msg %s"
+        stripped_payload = self.payloadWithSecretStripped(rpc)
+        logger.debug(debug_message, stripped_payload)
+        self.log_emitter("debug", debug_message % stripped_payload)
```

Comment:
Since that happen on every messages, I would not force the string formatting if no-one is listening.

Reply:
```python
if self.log_emitter.has("debug"):
    self.log_emitter.emit("debug", debug_message % stripped_payload)
```

Reply:
done